File: PathfinderAPI/src/main/java/com/github/ysl3000/bukkit/pathfinding/pathfinding/ControllerMove.java
package com.github.ysl3000.bukkit.pathfinding.pathfinding;

@Deprecated
public interface ControllerMove {

  boolean isOperationMove();

  void move(double motionX, double motionY, double motionZ, double speed);

  void move(float forward, float sideward);

  void update();


  double getX();

  double getY();

  double getZ();

  double getSpeed();


}

File: PathfinderAPI/src/main/java/com/github/ysl3000/bukkit/pathfinding/pathfinding/ControllerMove.kt
package com.github.ysl3000.bukkit.pathfinding.pathfinding

@Deprecated("")
interface ControllerMove {

    val isOperationMove: Boolean


    val x: Double

    val y: Double

    val z: Double

    val speed: Double

    fun move(motionX: Double, motionY: Double, motionZ: Double, speed: Double)

    fun move(forward: Float, sideward: Float)

    fun update()


}
